spec/services/digest_initiator_service_spec.rb

```diff
@@ -1,103 +1,48 @@
-        end
-      end
-
-      context "when a DigestRun already exists" do
-        it "doesn't create another one" do
-          create(:digest_run, :daily, date: Date.current)
+    before do
+      allow(DigestRunSubscriberQuery).to receive(:call).and_return(subscribers)
+      allow(DigestEmailGenerationWorker).to receive(:perform_async)
+    end
 
-          described_class.call(range: range)
+    context "when a digest run doesn't exist" do
+      it "creates a digest run for the current data with the specified range" do
+        expect { described_class.call(range: Frequency::DAILY) }
+          .to change { DigestRun.daily.where(date: Date.current).count }
+          .by(1)
 
-          expect(DigestRun.count).to eq(1)
-        end
+        expect { described_class.call(range: Frequency::WEEKLY) }
+          .to change { DigestRun.weekly.where(date: Date.current).count }
+          .by(1)
       end
 
-      context "when the service is called multiple times" do
-        it "only creates one DigestRun" do
-          described_class.call(range: range)
-          described_class.call(range: range)
-          described_class.call(range: range)
-          described_class.call(range: range)
-
-          expect(DigestRun.count).to eq(1)
+      it "marks the digest run as processed" do
+        freeze_time do
+          described_class.call(range: Frequency::DAILY)
+          digest_run = DigestRun.last
+          expect(digest_run.processed_at).to eq(Time.zone.now)
         end
       end
 
-      context "with matched content" do
-        let(:subscribers) { [create(:subscriber, id: 1), create(:subscriber, id: 2)] }
-
-        before do
-          allow(DigestRunSubscriberQuery).to receive(:call).and_return(subscribers)
-          allow(DigestEmailGenerationWorker).to receive(:perform_async)
-        end
-
-        it "creates a DigestRunSubscriber for each subscriber" do
-          described_class.call(range: range)
-          expect(DigestRunSubscriber.all.map(&:subscriber_id)).to match([1, 2])
-        end
-
-        it "enqueues a DigestEmailGenerationWorker job" do
-          expect(DigestEmailGenerationWorker)
-            .to receive(:perform_async)
-            .exactly(2).times
-
-          described_class.call(range: range)
-        end
+      it "creates a DigestRunSubscriber for each subscription" do
+        expect { described_class.call(range: Frequency::DAILY) }
+          .to change { DigestRunSubscriber.exists?(subscriber_id: subscribers.map(&:id)) }
+          .to(true)
       end
 
-      it "records a metric for the delivery attempt" do
-        expect(Metrics).to receive(:digest_initiator_service)
-          .with("daily")
-
-        described_class.call(range: range)
+      it "enqueus DigestEmailGenerationWorker for each DigestRunSubscriber" do
+        described_class.call(range: Frequency::DAILY)
+        expect(DigestEmailGenerationWorker).to have_received(:perform_async).twice
```

Comment:
Minor: we could/should check the arguments, like we do [here](https://github.com/alphagov/email-alert-api/blob/master/spec/workers/process_content_change_worker_spec.rb#L39).

Reply:
Sure, I've added something in for this. I didn't think it was too necessary really as the ids we test are created as part of the test but not too concerned about it being in there either.

Reply:
Ah, I was thinking more about the interface boundary: checking we're passing the arguments as expected (vs. the specific IDs).